src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -10755,6 +10755,9 @@ Real64 WaterThermalTankData::PlantMassFlowRatesFunc(EnergyPlusData &state,
         }
     }
 
+    Real64 const minFlow = 1.0e-20; // Catch underflow problems
+    if (FlowResult < minFlow) FlowResult = 0.0;
```

Comment:
This is a valid fix and would fix the problem and prevent the flow to take very small values.

Reply:
(This is beyond this one fix). I think there could be some global small-value defined so that it could also be used in some other places. 

I did a quick search in the code. There are a few places where a SMALL value of 1e-10 defined in some local functions such as in General.hh; but in most locations it is made only local to a function such as safe divisions: 
https://github.com/NREL/EnergyPlus/blob/cbd1174e5e3266057c2e5656fb1a06f9787f6ed2/src/EnergyPlus/General.cc#L909

Maybe making a few of such SMALL values global (with different names and values for different types of physics) might be helpful in the future? @RKStrand @mjwitte @Myoldmopar

Reply:
In DataHVACGlobals:

    Real64 constexpr SmallMassFlow(0.001);
    Real64 constexpr VerySmallMassFlow(1.0E-30);
    Real64 constexpr SmallWaterVolFlow(1.0E-9);

Reply:
Thanks @rraustad!  I'll change this to the VerySmallMassFlow and recommit.

Reply:
Thanks @rraustad and @RKStrand !

A side issue that I noticed when running a defect file is about the AIRLOOPHVAC:OUTDOORAIRSYSTEM transition program is that it cuts off the third line for a 3-line version--which calls a little bit treatment on the 3-line case. See #8995.

Reply:
@jcyuan2020 I have made a commit on this that changes to the existing VerySmallMassFlow constant.  So, this should be good to go.

Thanks for the heads-up on #8995.  I'm going to start working on that as soon as possible so that it can be done today.